vendor/bat-native-ads/src/bat/ads/internal/time_profiler.cc

```diff
@@ -0,0 +1,252 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
```

Comment:
BTW do we really need this custom profiling tool instead of standard chrome://tracing? 

Here is a chromium mechanism, which has a lot of benefits (tests, the views, IPC support for tracing renderer processes)
https://source.chromium.org/chromium/chromium/src/+/main:base/trace_event/common/trace_event_common.h;l=8

And here is an example using chrome://tracing to find performance bottlenecks:
https://source.chromium.org/chromium/chromium/src/+/main:docs/speed/debug-janks.md

Reply:
Temporary PR at this time as needed to profile iOS. This PR will not be merged until discussed at a future date, based upon bringing trace event support including UI to iOS